pulse_actions/handlers/builds.py

```diff
@@ -0,0 +1,44 @@
+import logging
+from mozci.mozci import trigger_missing_jobs_for_revision, trigger_range, trigger_all_talos_jobs
+
+logging.basicConfig(format='%(levelname)s:\t %(message)s')
+LOG = logging.getLogger()
+MEMORY_SAVING_MODE = True
+
+
+def on_builds_finish_event(data, message, dry_run):
+    # We need to ack the message to remove it from our queue
+    message.ack()
+    text = data['payload']['build']['text']
+    requester = data['payload']['build']['blame']
+    properties = data['payload']['build']['properties']
+    dictionary = {}
+    for property_list in properties:
+        if property_list[0] == 'branch':
+            repo_name = property_list[1]
+        if property_list[0] == 'revision':
+            revision = property_list[1]
+        if property_list[0] == 'buildername':
+            build = property_list[1]
+        if property_list[0] == 'mozci_request':
+            dictionary = property_list[1]
+
+    # if 'mozci_request' is not present, ignore the build and take no action
+    # or if the build has not completed successfully ignore.
+    if not dictionary or 'successful' not in text:
+        return
+
+    if dictionary['type'] == 'trigger_missing_jobs_for_revision':
+        LOG.info("Triggering missing jobs after build %s requested by %s has completed." % (build, requester))
+        trigger_missing_jobs_for_revision(repo_name, revision, dry_run=dry_run)
```

Comment:
We can just call trigger_missing_jobs functions in mozci - it takes ~2-3 minutes to go over all the jobs, and it won't trigger existing jobs (since there is already an instance), just the new jobs. Sending all the buildernames for a build again seemed to complicate things like maintaining a map.